--- fetchmail.c	2007-03-30 09:52:16.000000000 +0200
+++ fetchmail.c.oden	2008-12-20 19:17:12.000000000 +0100
@@ -163,7 +163,7 @@ int main(int argc, char **argv)
     envquery(argc, argv);
 #ifdef ENABLE_NLS
     setlocale (LC_ALL, "");
-    bindtextdomain(PACKAGE, LOCALEDIR);
+    bindtextdomain(PACKAGE, NULL);
     textdomain(PACKAGE);
     iana_charset = norm_charmap(nl_langinfo(CODESET)); /* normalize local
 							  charset to
